File: week_1/day-4-sliding-window/variations_que/averageSubarraysSizeK.js
const nums = [1,3,2,6,-1,4,1,8,2];
const k = 5;

function averages(nums, k) {
    let res = [];
    let sum = nums.slice(0, k).reduce((a,b)=>a+b,0);
    res.push(sum / k);

    for (let i = k; i < nums.length; i++) {
        sum += nums[i] - nums[i - k];
        res.push(sum / k);
    }
    return res;
}

console.log(averages(nums, k));

File: week_1/day-4-sliding-window/variations_que/average_subarrays_size_k.py
nums = [1, 3, 2, 6, -1, 4, 1, 8, 2]
k = 5

def averages(nums, k):
    res = []
    window_sum = sum(nums[:k])
    res.append(window_sum / k)

    for i in range(k, len(nums)):
        window_sum += nums[i] - nums[i - k]
        res.append(window_sum / k)

    return res

print(averages(nums, k))
